src/p_setup.c

```diff
@@ -1669,19 +1669,18 @@ void P_SetupLevel(int episode, int map, int playermask, skill_t skill)
     R_PrecacheLevel();
 
   // [FG] log level setup
-  {
-    I_Printf(VB_INFO, "P_SetupLevel: %.8s (%s), %s%s%s, %s complevel",
-      lumpname, W_WadNameForLump(lumpnum),
-      mapformat == MFMT_XNOD ? "XNOD nodes" :
-      mapformat == MFMT_ZNOD ? "ZNOD nodes" :
-      mapformat == MFMT_XGLN ? "XGLN nodes" :
-      mapformat == MFMT_ZGLN ? "ZGLN nodes" :
-      mapformat == MFMT_DEEP ? "DeepBSP nodes" :
-      "Doom nodes",
-      gen_blockmap ? " + Blockmap" : "",
-      pad_reject ? " + Reject" : "",
-      G_GetCurrentComplevelName());
-  }
+  I_Printf(VB_INFO, "P_SetupLevel: %.8s (%s), Skill %d, %s%s%s, %s (%d)",
+    lumpname, W_WadNameForLump(lumpnum),
+    gameskill + 1,
+    mapformat == MFMT_XNOD ? "XNOD" :
+    mapformat == MFMT_ZNOD ? "ZNOD" :
+    mapformat == MFMT_XGLN ? "XGLN" :
+    mapformat == MFMT_ZGLN ? "ZGLN" :
+    mapformat == MFMT_DEEP ? "DeepBSP" :
+    "Doom",
+    gen_blockmap ? "+Blockmap" : "",
+    pad_reject ? "+Reject" : "",
+    G_GetCurrentComplevelName(), demo_version);
```

Comment:
I think we don't need `demo_version`.

Reply:
What if we play a Boom 2.00 or 2.01 demo?

Reply:
> What if we play a Boom 2.00 or 2.01 demo?

Then we will desync, almost always. I think we should add a "unknown demo format" check for everything except 109, 111, 202, 203, 221 and 255.

Reply:
Just ran successfully through a Boom 2.01 demo:
```
S_ChangeMusic: D_EVIL (doom2.wad)
P_SetupLevel: MAP31 (GO2HELL2.WAD), Skill 4, Doom, Unknown (201)
G_DoPlayDemo: GO2HELL2 (lump)
S_ChangeMusic: D_DM2INT (doom2.wad)
```

Reply:
And Boom 2.00:
```
S_ChangeMusic: D_E3M9 (doom.wad)
P_SetupLevel: E3M9 (BAK2HELL.WAD), Skill 4, Doom, Unknown (200)
G_DoPlayDemo: BAK29NO (lump)
S_ChangeMusic: D_INTER (doom.wad)
```

Reply:
Okay, just found one that desyncs:
```
S_ChangeMusic: D_RUNNIN (AN_COOP2.WAD)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
G_DoPlayDemo: ANC2-855 (lump)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
P_SetupLevel: MAP01 (AN_COOP2.WAD), Skill 4, Doom, Unknown (201)
```

Reply:
So, "unknown" is probably appropriate enough.

Reply:
While at it, it should probably be very possible to achieve Boom 2.01 compatibility. 🤔

Reply:
> Okay, just found one that desyncs:

We have lost this ability somewhere, Woof 0.9 keeps sync! :sob: